### PR TITLE
fix(app): persist folder state in file selection drawer

### LINKED ISSUE #25416: directus_folders state gets lost in file interface
### Describe the Bug

We have a deeper folder structure and a collection with an M2A field (also tested with O2M). In the items of the sub-collection, there are file fields. When navigating in the file browser of the file interface, the state is not persistent when opening a different sub-item.

* The selected folder is lost.
* The expanded folders (to view subfolders) collapse.

In more complex folder structures, this becomes very time-consuming. I believe the state should be persistent for `directus_folders`. In the default file browser (`/admin/files`), the expanded folders seem to persist, but even there, I think the selected folder should also remain selected. It makes more sense than reverting to showing all files again when the user has already chosen a specific folder.

### To Reproduce

1. Create a multi-level folder structure for your files.
2. Create a collection **"posts"** with an **O2M field** to a collection **"blocks"**.
3. In the **"blocks"** collection, add a **file field**.
4. Add a new **"post"** and create a few **"blocks"** linked to this post.
5. Open **"block 1"**, navigate to a deeper level in the folder structure using the file interface, and close the drawers.
6. Now open **"block 2"** and open its file interface — all previously selected folders and expanded states are lost.

### Directus Version

v11.9.2

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

Postgres 16

### PR DESCRIPTION
**Pull request by costajohnt**
## Scope

What's changed:

- Added module-level state persistence in `drawer-files.vue` so folder navigation state (selected folder and special folder) survives drawer close/reopen cycles
- State is keyed by the `folder` prop to isolate different field configurations from each other

## Potential Risks / Drawbacks

- Module-level state persists for the entire browser session (not just the current drawer instance). This is intentional — the issue specifically requests that navigating to a subfolder should be remembered when reopening the drawer.
- Different file fields with different `folder` prop values get isolated state, so they won't interfere with each other.

## Tested Scenarios

- Open file selection drawer, navigate to a subfolder, close drawer, reopen → subfolder is preserved
- Open file selection drawer, switch to "My Files" special folder, close drawer, reopen → "My Files" is preserved
- Two different file fields with different root folder configurations maintain independent persisted state
- All 3 new unit tests pass; full test suite (105,397 tests) passes

## Review Notes / Questions

- The fix uses a `<script>` block alongside `<script setup>` for module-level state — this is a standard Vue 3 pattern for state that should survive component unmount/remount cycles
- The `setOpenFolders` watcher in `files-navigation.vue` automatically expands the folder tree to match the persisted `currentFolder`, so the open folders tree state is also correctly restored without changes to that file

## Checklist

- [x] Added or updated tests
- [x] Documentation PR created [here](https://github.com/directus/docs) or not required
- [x] OpenAPI package PR created [here](https://github.com/directus/openapi) or not required

---

Fixes #25416

### CODE REVIEW

File: app/src/views/private/components/drawer-files.test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		localOpenFolders: { type: Boolean, default: false },
+		actionsDisabled: { type: Boolean, default: false },
+	},
+	template: '<div class="files-nav-stub"></div>',
+});
+
+let global: GlobalMountOptions;
+let mockUserStore: any;
+
+beforeEach(() => {
```
Comment (HZooly) [inline app/src/views/private/components/drawer-files.test.ts:46]: The module-level `persistedState` Map isn't reset between tests. Test 1 (no `folder` prop → key `''`) writes `{ folder: 'subfolder-123' }` and leaves it. The next two tests use distinct keys so they pass, but any future test that mounts `DrawerFiles` without a `folder` prop will silently start with `subfolder-123` as the current folder.

A `beforeEach` that clears the Map (or `vi.resetModules()` + re-import) would make this deterministic. Alternatively, test 1 could use a unique folder key the way tests 2 and 3 do.

File: app/src/views/private/components/drawer-files.vue [hunk 2]
```diff
@@ -1,3 +1,9 @@
+<script lang="ts">
+// Persist folder navigation state across drawer open/close cycles (#25416).
+// Keyed by root folder prop to isolate state between different field configurations.
+const persistedState = new Map<string, { folder?: string; special?: SpecialFolder }>();
```
Comment (HZooly) [inline app/src/views/private/components/drawer-files.vue]: This lives for the entire module lifetime and is never cleared on logout. User A's folder navigation state will be visible to User B if they log in from the same tab. 

Also diverges from how the rest of the app handles cross-mount state — `nav-bar.ts`, `sidebar.ts`, and `use-expand-collapse.ts` all use `useLocalStorage`/`useSessionStorage` from `@vueuse/core`. Those composables accept a reactive key, which would let you scope state per user (`directus-drawer-files:{userId}:{folderKey}`) and avoid this problem entirely. Worth considering whether that pattern fits here.

> This is intentional — the issue specifically requests that navigating to a subfolder should be remembered when reopening the drawer.

I don't think the issue mentioned a need to persist between multiple sessions.

**Review by HZooly — CHANGES_REQUESTED** [review]

**Comment by HZooly** [comment]
Thank you @costajohnt for your work! It looks to solve the original issue and tests work well too.
Left a few notes on test isolation and a edge case with the localStorage persistence.

**Comment by costajohnt** [comment]
Thanks for the review! The latest commit swaps the module-level Map for useSessionStorage and adds sessionStore.clear() in beforeEach for test isolation.

On the per-user key scoping -- I looked at how nav-bar.ts, sidebar.ts, and use-expand-collapse.ts handle their keys and none of them include a userId either (just plain keys like `nav-bar-collapsed`, `sidebar-size`, etc). Since sessionStorage already clears on tab close, the cross-user scenario requires a same-tab logout/login, which those other stores are equally susceptible to. Kept our approach consistent with that existing pattern for now, but happy to add userId scoping if you think it's worth diverging from the convention.

File: app/src/views/private/components/drawer-files.vue [hunk 3]
```diff
@@ -26,15 +27,23 @@ const drawerProps = {
 	sidebarLabel: t('folders'),
 };
 
-const currentFolder = ref<string | undefined>(props.folder);
-const currentSpecial = ref<SpecialFolder>();
+const stateKey = props.folder ?? '';
+
+const persisted = useSessionStorage<{ folder?: string; special?: SpecialFolder }>(
+	`directus-drawer-files-state:${stateKey}`,
```
Comment (HZooly) [inline app/src/views/private/components/drawer-files.vue:36]: When `props.folder` is `undefined` (no root folder configured on the field), `stateKey` collapses to an empty string and all such instances share the key `directus-drawer-files-state:`. So two file fields on the same form that both lack a folder restriction will clobber each other's persisted navigation state navigating in one will corrupt the starting folder shown by the other.

Not a big issue, but I think we can isolate the state. That makes each field's state fully isolated regardless of whether a root folder is configured. A quick solution could be to use the `field`.

**Review by HZooly — CHANGES_REQUESTED** [review]
Thanks for the update! A very small last thing.

File: app/src/views/private/components/drawer-files.vue [hunk 4]
```diff
@@ -26,15 +28,25 @@ const drawerProps = {
 	sidebarLabel: t('folders'),
 };
 
-const currentFolder = ref<string | undefined>(props.folder);
-const currentSpecial = ref<SpecialFolder>();
+// Positional key so that field-only and folder-only contexts never collide
+// (e.g. field="avatar" → "avatar:", folder="abc" → ":abc", both → "avatar:abc")
+const stateKey = `${props.field ?? ''}:${props.folder ?? ''}`;
```
Comment (HZooly) [inline app/src/views/private/components/drawer-files.vue]:
```suggestion
const stateKey = `${props.collection ?? ''}:${props.field ?? ''}:${props.folder ?? ''}`;
```

What do you think of including the `collection` in the key? So if a field is present on 2 different collections, let say `thumbnail` field on `author` and `article` collections, we still can distinct them.

**Review by HZooly — CHANGES_REQUESTED** [review]
Last thing I missed in the previous review, after that I think we're good to go!

**Review by HZooly — APPROVED** [review]
Looks good to me, thanks for your contribution @costajohnt 🎊

**Review by AlexGaillard — APPROVED** [review]
Thanks @costajohnt, congrats on your first submission :cool::beans:
